pages/[id].js

```diff
@@ -0,0 +1,86 @@
+import { plants } from "@/assets/plants";
+import { useRouter } from "next/router";
+import Image from "next/image";
+import Link from "next/link";
+import styled from "styled-components";
+
+export default function PlantDetails() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  if (!router.isReady) return null;
+
+  const plant = plants.find((plant) => plant.id === id);
+
+  if (!plant) return <p>Plant not found</p>;
+
+  const lightNeedIcon = {
+    "Full Sun": "☀️☀️☀️☀️",
+    "Partial Shade": "☀️☀️☀️",
+    Shade: "☀️☀️",
+    "Full Shade": "☀️",
+  };
+
+  const waterNeedIcon = {
+    Low: "💧",
+    Medium: "💧💧",
+    High: "💧💧💧",
+  };
+
+  const seasonIcons = {
+    Spring: "🌱",
+    Summer: "🐝",
+    Fall: "🍂",
+    Winter: "❄️",
+  };
+
+  return (
+    <>
+      <Link href="/">
+        <button type="button">Back</button>
```

Comment:
please don't place a button inside a link. The link itself is a clickable element, so no need for an extra button.